spec/controllers/api/orders_controller_spec.rb

```diff
@@ -156,6 +158,109 @@ module Api
       end
     end
 
+    describe "#show" do
+      let!(:order) { create(:completed_order_with_totals, order_cycle: order_cycle, distributor: distributor ) }
+
+      context "Resource not found" do
+        before { allow(controller).to receive(:spree_current_user) { admin_user } }
+
+        it "when no order number is given" do
+          get :show, id: nil
+          expect_resource_not_found
+        end
+
+        it "when order number given is not in the systen" do
+          get :show, id: "X1321313232"
+          expect_resource_not_found
+        end
+
+        def expect_resource_not_found
+          expect(json_response).to eq( "error" => "The resource you were looking for could not be found." )
+          expect(response.status).to eq(404)
+        end
```

Comment:
isn't [have_http_status](https://relishapp.com/rspec/rspec-rails/docs/matchers/have-http-status-matcher) enough? line 178 feels like testing Rails to me and that's already pretty well tested.

Reply:
it's not testing rails, it's verifying that this controller plays well with the base_controller, but I agree it is not necessary here :+1:

Reply:
I use have_http_status  now :+1:
thanks!